### PR TITLE
Modify golangci.yaml file. Resolve found lint issues.

### LINKED ISSUE #6024: Update Velero golangci.yaml file and enable more linters.
**Describe the problem/challenge you have**

**Describe the solution you'd like**

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change
Remove the enabled linters setting from `Makefile`. Move the enabled linters list into `golangci.yaml`.
Enable more linters, including bodyclose, goheader, misspell, typecheck and unused.

Fix found issues found by linters.

# Does your change fix a particular issue?

Fixes #6024 

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/restore/priority.go [hunk 1]
```diff
@@ -26,9 +26,10 @@ const (
 )
 
 // Priorities defines the desired order of resource operations:
-//   Resources in the HighPriorities list will be handled first
-//   Resources in the LowPriorities list will be handled last
-//   Other resources will be handled alphabetically after the high prioritized resources and before the low prioritized resources
+//
+//	Resources in the HighPriorities list will be handled first
+//	Resources in the LowPriorities list will be handled last
+//	Other resources will be handled alphabetically after the high prioritized resources and before the low prioritized resources
```
Comment (qiuming-best) [inline pkg/restore/priority.go:32]: Is it ok with this indentation?

File: pkg/test/helpers.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (qiuming-best) [inline pkg/test/helpers.go:8]: Is it ok with this indentation?

File: pkg/test/mock_pod_command_executor.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (qiuming-best) [inline pkg/test/mock_pod_command_executor.go:8]: Is it ok with this indentation?

File: internal/restartabletest/restartable_delegate.go [hunk 4]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (qiuming-best) [inline internal/restartabletest/restartable_delegate.go:8]: Is it ok with this indentation?

File: pkg/restore/priority.go (hunk 1 above)
Comment (blackpiglet) [reply pkg/restore/priority.go]: Removed the tab.

File: internal/restartabletest/restartable_delegate.go (hunk 4 above)
Comment (blackpiglet) [reply internal/restartabletest/restartable_delegate.go]: Removed the tab.

**Review by qiuming-best — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**